pkg/workflow/task/bootstrap/bootstrap.go

```diff
@@ -2,37 +2,31 @@ package bootstrap
 
 import (
 	"context"
+	"fmt"
+	"io"
+	"os"
 
-	"github.com/aws/eks-anywhere/pkg/bootstrapper"
 	"github.com/aws/eks-anywhere/pkg/cluster"
-	"github.com/aws/eks-anywhere/pkg/constants"
 	"github.com/aws/eks-anywhere/pkg/types"
 	"github.com/aws/eks-anywhere/pkg/workflow/workflowcontext"
 )
 
-// OptionsRetriever supplies bootstrap cluster options. This is typically satisfied
-// by a provider.
-type OptionsRetriever interface {
-	BootstrapClusterOpts(*cluster.Spec) ([]bootstrapper.BootstrapClusterOption, error)
+// BootstrapCluster is an instance of a bootstrap cluster.
+type BootstrapCluster interface {
+	// CreateCluster creates the bootstrap cluster instance.
+	CreateCluster(context.Context) error
```

Comment:
nit: maybe just `Create`? Or something like `cluster.SpinUp`?
Same for delete. The cluster part seems a bit redundant given this is a cluster.

Reply:
We could do create. It felt like a bit of a weak interface doing that though. That said, I'm not sure `*Cluster` strengthens the interface 🤦🏻